### PR TITLE
Add router to dashboard

### PR DESCRIPTION
**Pull request by daurer**
Needs #269 

Uses a React router component to create multiple pages within the same dashboard app.

### CODE REVIEW

File: frontend/dashboard/package.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{
+  "name": "ptyhub",
```
Comment (iamvigneshwars) [inline frontend/dashboard/package.json:2]: Maybe the package name should be dashboard or workflows-dashboard?

File: frontend/dashboard/package.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    "dev": "vite",
+    "build": "relay-compiler --validate && tsc -b && vite build",
+    "lint": "eslint .",
+    "preview": "vite preview",
+    "relay": "relay-compiler"
+  },
+  "dependencies": {
+    "@dagrejs/dagre": "^1.1.4",
+    "@emotion/react": "^11.13.3",
+    "@emotion/styled": "^11.13.0",
```
Comment (iamvigneshwars) [inline frontend/dashboard/package.json:16]: Do we need these dependencies?

File: frontend/dashboard/package.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    "@jsonforms/material-renderers": "^3.4.1",
+    "@jsonforms/react": "^3.4.1",
+    "@mui/icons-material": "^6.1.6",
+    "@mui/material": "^6.1.6",
+    "@mui/system": "^6.1.6",
+    "@mui/x-date-pickers": "^7.22.1",
+    "@xyflow/react": "^12.3.4",
```
Comment (iamvigneshwars) [inline frontend/dashboard/package.json:23]: Do we need this dependencies?

File: frontend/dashboard/relay.config.json
Comment (iamvigneshwars) [inline frontend/dashboard/relay.config.json:1]: EOF

File: frontend/dashboard/src/App.css
Comment (iamvigneshwars) [inline frontend/dashboard/src/App.css:1]: can we remove these auto generated files?

File: frontend/dashboard/src/RelayEnvironment.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+const HTTP_ENDPOINT = "https://workflows.diamond.ac.uk/graphql";
+
+const kcinit = keycloak.init({
+  onLoad: "login-required"
+})
+.then(
+  auth => {
+    if (!auth) {
+      // window.location.reload();
```
Comment (iamvigneshwars) [inline frontend/dashboard/src/RelayEnvironment.ts:18]: Is there any reason this is commented out?

File: frontend/dashboard/src/errorpage.tsx
Comment (iamvigneshwars) [inline frontend/dashboard/src/errorpage.tsx:1]: EOF

File: frontend/dashboard/src/index.css
Comment (iamvigneshwars) [inline frontend/dashboard/src/index.css:1]: Can we remove these auto generated files?

File: frontend/dashboard/src/keycloak.ts
Comment (iamvigneshwars) [inline frontend/dashboard/src/keycloak.ts:1]: EOF

File: frontend/dashboard/src/keycloak.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import Keycloak from "keycloak-js";
+
+//Keycloak init options
+const initOptions = {
+  url: "https://authn.diamond.ac.uk/",
+  realm: "master",
+  clientId: "iat69393",
```
Comment (iamvigneshwars) [inline frontend/dashboard/src/keycloak.ts:7]: Shouldn't we have a generic `clientId` ?

**Review by iamvigneshwars — CHANGES_REQUESTED** [review]

File: frontend/dashboard/src/keycloak.ts (hunk 5 above)
Comment (daurer) [reply frontend/dashboard/src/keycloak.ts:7]: Yeah, should probably add a workflows clientId for testing purposes

File: frontend/.devcontainer/devcontainer.json [hunk 6]
```diff
@@ -5,7 +5,8 @@
     "ghcr.io/devcontainers/features/common-utils:2.4.2": {
       "username": "none",
       "upgradePackages": false
-    }
+    },
+    "git-lfs": "latest"
```
Comment (garryod) [inline frontend/.devcontainer/devcontainer.json:9]: Why do we need `lfs`?

File: frontend/dashboard/src/routes/workflows.tsx
Comment (garryod) [inline frontend/dashboard/src/routes/workflows.tsx:1]: The weird spacing is back (not actually an issue, just found it amusing)

**Review by garryod — COMMENTED** [review]
Looks good, just one query

File: frontend/.devcontainer/devcontainer.json (hunk 6 above)
Comment (garryod) [reply frontend/.devcontainer/devcontainer.json:9]: Discussed offline, seems reasonable. Could we use the fully qualified syntax? I.e.:
```suggestion
    },
    "ghcr.io/devcontainers/features/git-lfs:1.2.3": {}
```

**Review by iamvigneshwars — APPROVED** [review]
LGTM

**Pull request merged**